base/array.jl

```diff
@@ -879,6 +879,18 @@ function lexcmp(a::Array{UInt8,1}, b::Array{UInt8,1})
     return c < 0 ? -1 : c > 0 ? +1 : cmp(length(a),length(b))
 end
 
+# use memcmp for == on bit integer types
+=={T<:BitInteger,N}(a::Array{T,N}, b::Array{T,N}) =
+    size(a) == size(b) &&
+    ccall(:memcmp, Int32, (Ptr{T}, Ptr{T}, UInt), a, b, sizeof(T) * length(a)) == 0
```

Comment:
I find it more readable to lead with the `0 ==` as opposed to having to match parens over the entire length of the line to see where the ccall starts and ends. Also if one of these is using `function` form, they're similar enough that may as well use the same form in both.